include/swift/AST/DiagnosticsSema.def

```diff
@@ -1391,6 +1391,10 @@ ERROR(pattern_binds_no_variables,none,
       "variables",
       (unsigned))
 
+WARNING(ambiguous_enum_case,none, "enum case '%0' exists in both '%1' and "
+      "'Optional', using the one from Optional",
```

Comment:
It might be better to phrase this more like "assuming you mean 'Optional.none'; did you mean '%1' instead?"; it'd be a little less intimidating.